src/core_modules/capture-core/hooks/useTrackedEntityTypesWithCorrelatedPrograms.js

```diff
@@ -32,9 +33,28 @@ export const useTrackedEntityTypesWithCorrelatedPrograms = (): TrackedEntityType
                 },
                 searchGroups,
                 enrollment,
+                stages,
+                useFirstStageDuringRegistration,
             }: TrackerProgram) => {
                 const accumulatedProgramsOfTrackedEntityType =
                   acc[trackedEntityTypeId] ? acc[trackedEntityTypeId].programs : [];
+                const programMetadata = { programId, programName, searchGroups, enrollment };
+                if (useFirstStageDuringRegistration) {
+                    const accessibleStages = [...stages.values()].filter(({ access }) => access.write);
+                    if (accessibleStages.length) {
+                        const firstAccessibleStage = accessibleStages[0];
+                        firstAccessibleStage.stageForm.getSection(Section.MAIN_SECTION_ID).name
```

Comment:
This breaks if the data entry form of the first stage is a section form. We should probably still allow different form types. IMO we should also not overwrite the section header if this is custom. (But I agree that the design and UI seems like this is intended)